Frameworks/UIKit/NSLayoutManager.mm

```diff
@@ -72,105 +85,107 @@ - (void)__layoutAllText {
         _frame.reset(CTFramesetterCreateFrame(framesetter.get(), {}, path.get(), nullptr));
     }
 
+    NSTextContainer* container = (NSTextContainer*)_textContainers[0];
+    const CGSize containerSize = container.size;
+    CGPoint origin{ 0, 0 };
+
     for (id lineRef in static_cast<NSArray*>(CTFrameGetLines(_frame.get()))) {
```

Comment:
If you're just using the WOCCatalog stuff to test this, may want to mix up the text by tossing in some newlines to make sure those are handled well.  Everything here looks like it will work better, but I'm not sure if there are any edge cases snuck in.